py/desitarget/secondary.py

```diff
@@ -576,20 +576,42 @@ def match_secondary(primtargs, scxdir, scndout, sep=1.,
     # APC obsconditions for specific DESI_TARGET bits
     # APC See https://github.com/desihub/desitarget/pull/530
 
-    # APC Only consider primary targets with secondary bits set
-    scnd_update = (targs[desicols[0]] & desi_mask['SCND_ANY']) != 0
-    if np.any(scnd_update):
-        # APC Allow changes to primaries if the DESI_TARGET bitmask has
-        # APC only the following bits set, in any combination.
-        log.info('Test if secondaries can update {} matched primaries'.format(
-            scnd_update.sum()))
-        update_from_scnd_bits = (desi_mask['SCND_ANY'] | desi_mask['MWS_ANY'] |
-                                 desi_mask['STD_BRIGHT'] | desi_mask['STD_FAINT']
-                                 | desi_mask['STD_WD'])
-        scnd_update &= ((targs[desicols[0]] & ~update_from_scnd_bits) == 0)
-        log.info('New priority, numobs, obscon for {} matched primaries'.format(
-            scnd_update.sum()))
+    # APC Default behaviour is that targets with SCND_ANY bits set will ONLY be
+    # APC have initial state set based on their secondary targetmask parameters IF
+    # APC they have NO primary target bits set (hence == on next line).
+    scnd_update = targs[desicols[0]] == desi_mask['SCND_ANY']
+    log.info('{} scnd targets will have initial state set as secondary-only'.format(scnd_update.sum()))
+
+    # APC The exception to the rule above is that a subset of bits flagged with
+    # APC updatemws=True in the targetmask can drive initial state for a subset of
+    # APC primary bits corresponding to MWS targets and standards. We first create
+    # APC a bitmask of those permitted seconday bits.
+    permit_scnd_bits = 0
+    for name in scnd_mask.names():
```

Comment:
Is this backwards compatible? For instance, if we needed to run the `sv3` version of MTL again, will this fail because the `sv3` bit-masks have no concept of the `updatemws` attribute?

It seems to me that you might need an `if survey=="main"` clause somewhere?

Reply:
I've added logic to only use `updatemws` if `survey==main`. Otherwise all bits are permitted to drive updates to MWS targets as before this PR.